parsl/executors/workqueue/exec_parsl_function.py

```diff
@@ -0,0 +1,183 @@
+    result_name = prefix + "result"
+
+    # Add variables to the namespace to make function call
+    user_namespace.update({args_name: fn_args,
+                           kwargs_name: fn_kwargs,
+                           result_name: result_name})
+
+    if isinstance(fn, str):
+        code = encode_source_code_function(user_namespace, fn, fn_name, args_name, kwargs_name, result_name)
+    elif callable(fn):
+        code = encode_byte_code_function(user_namespace, fn, fn_name, args_name, kwargs_name, result_name)
+    else:
+        raise ValueError("Function object does not look like a function.")
+
+    return (code, result_name)
+
+
+def encode_source_code_function(user_namespace, fn, fn_name, args_name, kwargs_name, result_name):
+    # We drop the first line as it names the parsl decorator used (i.e., @python_app)
+    source = fn.split('\n')[1:]
+    fn_app = "{0} = {1}(*{2}, **{3})".format(result_name, fn_name, args_name, kwargs_name)
+
+    source.append(fn_app)
+
+    code = "\n".join(source)
+    return code
+
+
+def encode_byte_code_function(user_namespace, fn, fn_name, args_name, kwargs_name, result_name):
+    user_namespace.update({fn_name: fn})
+    code = "{0} = {1}(*{2}, **{3})".format(result_name, fn_name, args_name, kwargs_name)
+    return code
+
+
+def execute_function(map_file, function_file):
+    # Get all variables from the user namespace, and add __builtins__
+    user_ns = locals()
+    user_ns.update({'__builtins__': __builtins__})
+
+    function_info = load_pickled_file(function_file)
+
+    (fn, fn_name, fn_args, fn_kwargs) = unpack_function(function_info, user_ns)
+
+    mapping = load_pickled_file(map_file)
+    remap_all_files(mapping, fn_args, fn_kwargs)
+
+    (code, result_name) = encode_function(user_ns, fn, fn_name, fn_args, fn_kwargs)
+
+    exec(code, user_ns, user_ns)
+    result = user_ns.get(result_name)
+
+    return result
+
+
+if __name__ == "__main__":
+    try:
+        # parse the three required command line arguments:
+        # map_file: contains a pickled dictionary to map original names to
+        #           names at the execution site.
+        # function_file: contains the pickled parsl function to execute.
+        # result_file: any output (including exceptions) will be written to
+        #              this file.
+        try:
+            (map_file, function_file, result_file) = sys.argv[1:]
+        except ValueError:
+            print("Usage:\n\t{} function result mapping\n".format(sys.argv[0]))
+            raise
+
+        result = execute_function(map_file, function_file)
+    except Exception:
+        print("There was an error while setting up the function.")
+        traceback.print_exc()
+        result = RemoteExceptionWrapper(*sys.exc_info())
+
+    # Write out function result to the result file
+    try:
+        dump_result_to_file(result_file, result)
+    except Exception:
+        print("Could not write to result file.")
+        traceback.print_exc()
```

Comment:
Should this exception clause make the process with a non-zero exit code? I think this `except` clause falls off the end and lets the process exit normally with a zero exit code.

Reply:
Fixed in e5a1aea50e1a4c2f520e6dc02e86d57e6f4e1db5